archive/walk.go

```diff
@@ -29,10 +29,11 @@ const (
 type FileKind string
 
 const (
-	FileKindUnknown  FileKind = ""
-	FileKindData              = "d"
-	FileKindMetadata          = "m"
-	FileKindSeek              = "ts"
+	FileKindUnknown       FileKind = ""
+	FileKindData                   = "d"
+	FileKindDataCompacted          = "dc"
```

Comment:
Is it important to create a file kind for compacted vs not-compacted? Are they not both just data files?

Reply:
Good question: It is no longer needed to differentiate them, and I'll make that change.

I started this work before #1528, and before that change, we needed to differentiate between the two cases so that if walk obtained a file with compacted data, it could look at the file type and know it must look at the metadata to get the masks list. Now that we are always looking at the metadata for the chunk, there's no need (I can think of) to differentiate between them.